terraform/modules/coreos_asg/main.tf

```diff
@@ -0,0 +1,223 @@
+data "aws_region" "selected" {
+    current = true
+}
+
+data "aws_caller_identity" "current" {}
+
+resource "aws_ecs_cluster" "cluster" {
+    name = "${var.name}"
+}
+
+data "aws_ami" "coreos_stable" {
```

Comment:
@KellerFuchs droneci has the ability to deploy k8s jobs, apply terraform configuration on merge to master, check signatures, build containers, and do pretty much anything we want to automate via the many available modules. The drone.yml files included in projcets are ~ a superset of travis.

That is why I opted to put this in a dedicated VPC/Enviornment with full machine/network isolation away from the production environments.

If we were to later have a develop environment or a stanging enviornment etc we would allow this to write to any environment, without giving them access to talk to each other.

Also since this module will be managing terraform templates, it can't manage itself via terraform. Given that this enviornment has a dedicated terraform root-config to manage drone and all the AWS resources it needs to function.

From there, it could then manage any other VPCs/environments via other terraform configs.

Really shooting for this to be very hands-off thus the auto-dns updating etc so it can just terminate and rebuild itself on failure or for updates.